src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ProjectDebuggerProvider.cs

```diff
@@ -0,0 +1,338 @@
+using Microsoft.VisualStudio.ProjectSystem.Debug;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Debug
+{
+
+    /// <summary>
+    /// The exported CPS debugger for all types of K projects (web, consoles, classlibraries). Defers to
+    /// other types to get the DebugTarget information to launch.
+    /// </summary>
+    [ExportDebugger(ProjectDebugger.SchemaName)]
+    [AppliesTo(ProjectCapability.LaunchProfiles)]
+    internal class ProjectDebuggerProvider : DebugLaunchProviderBase, IDeployedProjectItemMappingProvider
+    {
+
+        /// <summary>
+        /// Constructors. Unit test one is 2nd
+        /// </summary>
+        [ImportingConstructor]
+        public ProjectDebuggerProvider(ConfiguredProject configuredProject, ILaunchSettingsProvider launchSettingsProvider)
+            : base(configuredProject)
+        {
+            LaunchSettingsProvider = launchSettingsProvider;
+
+            // We want it sorted so that higher numbers come first (is the default for these collections but explicitly expressed here)
+            ProfileLaunchTargetsProviders = new OrderPrecedenceImportCollection<IDebugProfileLaunchTargetsProvider>(ImportOrderPrecedenceComparer.PreferenceOrder.PreferredComesFirst, 
+                                                                                                                    configuredProject.UnconfiguredProject);
+        }
+
+        public ProjectDebuggerProvider(ConfiguredProject configuredProject, ILaunchSettingsProvider launchSettingsProvider, 
+                                       OrderPrecedenceImportCollection<IDebugProfileLaunchTargetsProvider> providers)
+            : base(configuredProject)
+        {
+            ProfileLaunchTargetsProviders = providers;
+            LaunchSettingsProvider = launchSettingsProvider;
+        }
+
+        /// <summary>
+        /// Import the LaunchTargetProviders which know how to run profiles
+        /// </summary>
+        [ImportMany]
+        private OrderPrecedenceImportCollection<IDebugProfileLaunchTargetsProvider> ProfileLaunchTargetsProviders { get; set; }
+
+        private ILaunchSettingsProvider LaunchSettingsProvider { get; }
+
+        // Tracks the last launched provider so we can forward calls to IDeployedProjectItemMappingProvider
+        public IDebugProfileLaunchTargetsProvider LastLaunchProvider { get; private set; }
+
+        /// <summary>
+        /// Called by CPS to determine whether we can launch
+        /// </summary>
+        public override Task<bool> CanLaunchAsync(DebugLaunchOptions launchOptions)
+        {
+            return TplExtensions.TrueTask;
+        }
+
+        /// <summary>
+        /// Helper returns the correct debugger engine based on the targeted framework
+        /// </summary>
+        public static Guid GetDebugEngineForFramework(string targetFramework)
```

Comment:
I would suggest `GetManagedDebugEngineGuid` and omit `ForFramework` because it is implicit from the method signature. If I am developer searching for an API in `ProjectDebuggerProvider` which returns Guid, my natural instinct would be `.` and type Guid to look for APIs that concerns Guid and I will be able to get to the APIs without having to probing those all the methods.